### ISSUE TITLE
chunkserver feature: automark volume for removal on errors

### ISSUE THREAD

**Issue by onlyjob** [lizardfs/lizardfs#239]
I'd like to propose configurable feature to automatically mark volumes for removal on write error(s). This will help to isolate malfunctioning HDDs quickly. As I've just painfully discovered, client and chunkserver keep trying to save chunk(s) many times to the same HDD despite errors -- this is quite noticeable on client side. Such situation can occur when corrupted file system auto-switched to read-only mode or with other types of file system problems. Besides it looks like a good idea to stop writing data to problematic HDD and allow safe re-distribution of its chunks while HDD is still online before it completely knocked down and became unavailable.

Maybe a counter like number of errors before auto-mark for removal? Also it is important to mark volume for removal only in chunkserver runtime without modifying config file. Thanks.

**Comment by marcinsulikowski** [comment]
This is probably because of a bug. We can see that it is very old -- inherited by LizardFS from MooseFS 1.6.27.

**Comment by marcinsulikowski** [comment]
Lack of automarking "for removal" is not a bug. A bug (fixed in f7335ba2edace8e483b4f7d02ffecbc5f75b5303) might have caused other disk-error related issues, but automarking "for removal" has never been implemented. But automarking as "damaged" is implemented and should work if a disk remounted readonly. If it still doesn't work -- it may be some other bug.

**Comment by onlyjob** [comment]
Originally I experienced this problem on chunkserver with Btrfs backend. Near-full Btrfs partition manifested this condition (that can be fixed by "btrfs balance") when some writes were failing with ENOSPC (No space left on device). Chunkserver was still trying to write data despite over ~80 errors until I manually disconnected failing HDD. I will test f7335ba but I no longer use Btrfs on my chunkservers...
